OpenDreamShared/Dream/DreamFilter.cs

```diff
@@ -0,0 +1,95 @@
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+
+namespace OpenDreamShared.Dream {
+    [Serializable, NetSerializable]
+    public sealed class DreamFilter : IEquatable<DreamFilter> {
+        public bool used = false; //this may not be the best way of doing this
+        [ViewVariables] public string? filter_type = null;
+        [ViewVariables] public float? filter_x = null;
+        [ViewVariables] public float? filter_y = null;
+        [ViewVariables] public Object? filter_icon = null; //todo what is this?
+        [ViewVariables] public Object? filter_render_source = null; //todo what is this?
+        [ViewVariables] public float? filter_flags = null;
+        [ViewVariables] public float? filter_size = null;
+        [ViewVariables] public string? filter_color_string = null;
+        [ViewVariables] public string? filter_threshold_color = null;
+        [ViewVariables] public float? filter_threshold_strength = null;
+        [ViewVariables] public float? filter_offset = null;
+        [ViewVariables] public float? filter_alpha = null;
+        [ViewVariables] public Object? filter_color_matrix = null; //matrix type?
+        [ViewVariables] public float? filter_space = null;
+        [ViewVariables] public Object? filter_transform = null; //matrix type?
+        [ViewVariables] public float? filter_blend_mode = null;
+        [ViewVariables] public float? filter_density = null;
+        [ViewVariables] public float? filter_factor = null;
+        [ViewVariables] public float? filter_repeat = null;
+        [ViewVariables] public float? filter_radius = null;
+        [ViewVariables] public float? filter_falloff = null;
```

Comment:
All these vars end up with lots of verbose & repeated code. Would it be possible to instead have a dictionary of keys/values for shader parameters?